snapcraft/pluginhandler.py

```diff
@@ -169,12 +169,15 @@ def _load_code(self, plugin_name, properties, part_schema):
 
         plugin = _get_plugin(module)
         options = _make_options(part_schema, properties, plugin.schema())
+        # For backwards compatibility we add the project options as an option
+        setattr(options, 'project', self._project_options)
```

Comment:
`options` currently have a pretty well-defined meaning: "options from the YAML." I like not breaking API in this way, but I don't like diluting `options`. Particularly because: what happens when a local plugin declares `project` as a property in the schema?

Would it be better to actually set a `__project` attribute on the plugin instance and add a getter method on the base plugin (or something similar)?